modules/apps/foundation/address-formatter/address-formatter-api/src/main/java/com/liferay/address/formatter/util/AddressFormatterUtil.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.address.formatter.util;
+
+import com.liferay.address.formatter.AddressFormatter;
+import com.liferay.osgi.service.tracker.collections.map.ServiceTrackerMap;
+import com.liferay.osgi.service.tracker.collections.map.ServiceTrackerMapFactory;
+import com.liferay.portal.kernel.model.Address;
+import com.liferay.portal.kernel.model.Country;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+
+/**
+ * @author Pei-Jung Lan
+ */
+@Component(immediate = true, service = AddressFormatterUtil.class)
+public class AddressFormatterUtil {
+
+	public static String format(Address address) {
+		Country country = address.getCountry();
+
+		String a2 = country.getA2();
+
+		AddressFormatter addressFormatter = getAddressFormatter(a2);
+
+		return addressFormatter.format(address);
+	}
+
+	public static AddressFormatter getAddressFormatter(String countryA2) {
```

Comment:
If it's limited to simply decide the format based on the country, what other factor should we take into account? Say if we want to take locale into consideration, does it make sense to display an address written in Japanese with an Spanish format to a Spainish user? Personally I think if the user can't read Japanese, the format won't make much a difference to him/her. All that the user needs to know is that it's an address in Japan. And that's why we have the country field localized. I don't think it's a common practice to show a country's address using another country's formatting rules or write an address in a different language because address fields are names, and names are hard to translate. However there're exceptions since some languages like Japan and China can be Romanized. This makes me start to think if we should just provide a util that checks if an address is romanized, and if so, it will be formatted using the Romanization (US) rules.

You brought up a good point. I totally missed it. I believe we've talked about this while gathering the requirements from Sam Kong and support. It makes sense to provide both the site and user's locale since the address can be displayed on a public page where there's no signed in user. Under that circumstance, we'd want to display the country in the site's locale instead of user's. There might be other use cases.

When you say Formatters overlapping with each other, are you referring to the reference of other formatters in a formatter (ie. using US formatter in the Japan formatter)? I don't see any issues with this. Am I missing something here?